src/bindgen/parser.rs

```diff
@@ -563,6 +572,160 @@ impl Parse {
+
+                self.assoc_types.insert(key.clone(), (conc_type.clone(), 0));
+
+                // Check if concrete type is another associated type
+                self.check_nested_assoc_types(&key, &mut conc_type);
+                println!("\nPending:\n {:#?}\n", self.pending_assoc_types);
+                println!("\nCurrent:\n {:#?}\n", self.assoc_types);
+
+                // Check if fully resolved, if so, check whether other types depend on this key
+                if let Some((resolved_ty, 0)) = self.assoc_types.get(&key) {
+                    self.check_satisfy(&key, &resolved_ty.clone());
+                }
+            }
+        }
+    }
+
+    // Recursive function, marks all nested associated types and count dependencies
+    fn check_nested_assoc_types(&mut self, key: &AssocTypeId, conc_type: &mut Type) {
+        match conc_type {
+            // Recursive calls for nested types
+            Type::Ptr { ty: ty_, .. } => {
+                self.check_nested_assoc_types(key, &mut **ty_);
+            }
+            Type::Array(ty_, _) => {
+                self.check_nested_assoc_types(key, &mut **ty_);
+            }
+            Type::FuncPtr { ret, args, .. } => {
+                self.check_nested_assoc_types(key, &mut **ret);
+                for (_, ty_) in args {
+                    self.check_nested_assoc_types(key, ty_);
+                }
+            }
+            // Base case, found a GenericPath with associated types
+            Type::Path(generic_path) => {
+                if let Some(id) = generic_path.assoc() {
+                    // Check if already resolved
+                    if let Some((ty, 0)) = self.assoc_types.get(id) {
+                        *conc_type = ty.clone();
+                    }
+                    // Unresolved nested associated types
+                    else {
+                        // Add unresolved key to pending map
+                        self.pending_assoc_types
+                            .entry(id.clone())
+                            .and_modify(|id_vec| {
+                                id_vec.push(key.clone());
+                            })
+                            .or_insert(vec![key.clone()]);
+
+                        // Increment self count
+                        self.assoc_types
+                            .entry(key.clone())
+                            .and_modify(|(_, count)| {
+                                *count += 1;
+                            });
+                    }
+                }
+            }
+            _ => {}
+        }
+    }
+
+    // Given key k that is fully resolved, check what depends on k and resolve those
+    fn check_satisfy(&mut self, key: &AssocTypeId, resolved_ty: &Type) {
+
+        if let Some(id_vec) = self.pending_assoc_types.get(key) {
+            // Resolve for each dependent key
+            for id in id_vec {
+                let (curr_ty, count) = self.assoc_types.get_mut(id).unwrap();
```

Comment:
```suggestion
                let (curr_ty, count) = &mut self.assoc_types[id];
```

Reply:
Compiler complains about this. <trait `IndexMut` is required to modify indexed content>. It suggests to use get_mut(), insert() or the Entry API

Reply:
interesting, my bad